api/app/src/main/kotlin/packit/security/oauth2/OAuth2SuccessHandler.kt

```diff
@@ -0,0 +1,47 @@
+package packit.security.oauth2
+
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.security.core.Authentication
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler
+import org.springframework.stereotype.Component
+import org.springframework.web.util.UriComponentsBuilder
+import packit.AppConfig
+import packit.security.issuer.JwtIssuer
+
+@Component
+class OAuth2SuccessHandler(
+    val config: AppConfig,
+    val jwtIssuer: JwtIssuer,
+) : SimpleUrlAuthenticationSuccessHandler()
+{
+    override fun onAuthenticationSuccess(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        authentication: Authentication,
+    )
+    {
+        handle(request, response, authentication)
+        super.clearAuthenticationAttributes(request)
+    }
+
+    override fun handle(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        authentication: Authentication,
+    )
+    {
+        var targetUrl =
+            config.authRedirectUri.ifEmpty { determineTargetUrl(request, response, authentication) }
+
+        val token = jwtIssuer.issue(authentication)
```

Comment:
So we still want to use a JWT token for OAuth2 users (encodes roles etc) but that token  will be signed with the secret we configured for basic auth? 
https://github.com/mrc-ide/packit/pull/29/files#diff-3cfc73e7540bb99e4da3587ab2e12ea5eb950c8fc6b92dae45dc99305717e8ccR40

Reply:
Yes, we are issuing the token because app is stateless.